aviatrix/resource_aviatrix_kubernetes_cluster.go

```diff
@@ -0,0 +1,285 @@
+				Optional:    true,
+				Description: "AWS ARN of the cluster if it is an EKS cluster.",
+				ValidateDiagFunc: func(i interface{}, path cty.Path) diag.Diagnostics {
+					arn := i.(string)
+					if _, _, _, err := parseArn(arn); err != nil {
+						return diag.Errorf("invalid ARN: %s", err)
+					}
+					return nil
+				},
+			},
+			"use_csp_credentials": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Description: "Whether to use the credential of the account.",
+			},
+			"kube_config": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "Kube config file content of the cluster.",
+				Sensitive:   true,
+				DiffSuppressFunc: func(k, old, new string, d *schema.ResourceData) bool {
+					// The controller trims all encrypted content, so we need to trim the kube config content before comparing
+					oldTrimmed := strings.TrimSpace(old)
+					newTrimmed := strings.TrimSpace(new)
+					return oldTrimmed == newTrimmed
+				},
+			},
+			"account_name": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Description:  "Account name owning the cluster.",
+				RequiredWith: []string{"account_id", "name", "vpc_id", "region", "version", "platform", "public", "network_mode"},
+			},
+			"account_id": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Computed:     true,
+				Description:  "Account ID owning the cluster.",
+				RequiredWith: []string{"account_name", "name", "vpc_id", "region", "version", "platform", "public", "network_mode"},
+			},
+			"name": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Description:  "Display name of the cluster",
+				RequiredWith: []string{"account_id", "account_name", "vpc_id", "region", "version", "platform", "public", "network_mode"},
+			},
+			"vpc_id": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Description:  "Id of the VPC where the cluster is deployed.",
+				RequiredWith: []string{"account_id", "account_name", "name", "region", "version", "platform", "public", "network_mode"},
+			},
+			"region": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Description:  "Region where the cluster is deployed.",
+				RequiredWith: []string{"account_id", "account_name", "name", "vpc_id", "version", "platform", "public", "network_mode"},
+			},
+			"version": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Description:  "Version of the Kubernetes cluster.",
+				RequiredWith: []string{"account_id", "account_name", "name", "vpc_id", "region", "platform", "public", "network_mode"},
+			},
+			"platform": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Description:  "Platform of the Kubernetes cluster, e.g. kops, kubeadm or any other free string.",
+				RequiredWith: []string{"account_id", "account_name", "name", "vpc_id", "region", "version", "public", "network_mode"},
+			},
+			"public": {
```

Comment:
I feel like 
`is_public` 
or
`publicly_accessible`

better denote that is is a bool

Reply:
Thank you! I renamed the field to `is_publicly_accessible`.